commit 7e9d036a35050d7b6d36710162681ae93cf28ec2
Author:     Yuan-Ting Hsieh (謝沅廷) <yuantingh@nvidia.com>
AuthorDate: Wed Oct 5 14:58:50 2022 -0700
Commit:     GitHub <noreply@github.com>
CommitDate: Wed Oct 5 14:58:50 2022 -0700

    Fix server engine abort_app_on_server (#989)
    
    * Fix server engine
    
    * Address comment
---
 nvflare/private/fed/server/server_engine.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/nvflare/private/fed/server/server_engine.py b/nvflare/private/fed/server/server_engine.py
index 50806de8..3199e00d 100644
--- a/nvflare/private/fed/server/server_engine.py
+++ b/nvflare/private/fed/server/server_engine.py
@@ -344,7 +344,6 @@ class ServerEngine(ServerEngineInternalSpec):
 
         self.logger.info("Abort the server app run.")
 
-        status_message = ""
         try:
             status_message = self.send_command_to_child_runner_process(
                 job_id=job_id,
@@ -359,10 +358,11 @@ class ServerEngine(ServerEngineInternalSpec):
                     child_process.terminate()
         finally:
             with self.lock:
-                self.run_processes.pop(job_id)
+                if job_id in self.run_processes:
+                    self.run_processes.pop(job_id)
 
         self.engine_info.status = MachineStatus.STOPPED
-        return status_message
+        return ""
 
     def check_app_start_readiness(self, job_id: str) -> str:
         if job_id not in self.run_processes.keys():
